Geometry/VeryForwardGeometryBuilder/src/DetGeomDesc.cc

```diff
@@ -47,14 +47,14 @@ DetGeomDesc::DetGeomDesc(const cms::DDFilteredView& fv, const bool isRun2)
     : m_name(computeNameWithNoNamespace(fv.name())),
       m_copy(fv.copyNum()),
       m_isDD4hep(true),
-      m_trans(geant_units::operators::convertCmToMm(fv.translation())),  // converted from cm (DD4hep) to mm
+      m_trans(fv.translation() / dd4hep::mm),  // converted from cm (DD4hep) to mm
       m_rot(fv.rotation()),
       m_params(computeParameters(fv)),  // default unit from DD4hep (cm)
```

Comment:
I would convert `m_params` to mm right here (actually in `computeParameters`). Then all members of `DetGeomDesc` would be in Geant4 units. `computeDiamondDimensions` could then be simplified because the input units would be always mm.

Reply:
When old DD is removed, that case will disappear anyway.
In the meantime, I would not make old DD case depend on the DD4hep units, while the DD4hep case needs the `/dd4hep::mm` to have the certainty of getting mm regardless of the DD4hep internals.

Reply:
@ghugo83 I am saying the opposite. Make the interface consistent. For DetGeomDesc members and methods, all units should be mm. Right now, the `params` method (which is never used, as far as I can tell) and `m_params` store DD4hep units, while all the other members and methods use mm.
If `m_params` in the DD4hep version were converted to mm at initialization, then `computeDiamondDimensions` would expect mm only, and it could be simplified by the removal of the `if`.

Reply:
@cvuosalo Ha yes, I looked at it more properly, sorry I see what you mean now.
params() is used (from legacy) in CondTools/Geometry/plugins/PGeometricDetBuilder.cc to store PGeometricDet info to DB.
It is true that it would be easier to just convert all shape parameters to mm in a computeParameters function. The computeDiamondDimensions would just be a particular case of this, in the event that the shape is a box.

It is actually the version which I had at first. 
But the thing is that is was implying **a specific treatment for each possible shape**, because we want to convert **only the parameters which are lengths, and not all the parameters of the shape.**
This was making the computeParameters() function quite heavy.
And in practice, as you noticed, only the box shape parameters are used anyway in the PPS legacy code.

Hence, after discussion with the PPS team, it was decided that the shape parameters are only treated in the case of a box, with the computeDiamondDimensions and getDiamondDimensions functions.
The params() function instead, which provides info to DB, is kept untouched, and just provides the shape parameters in their native format (units, but also orders of parameters).

Shall we want to simplify this, we could just remove params() function - but that would imply removing (useless) info stored in DB in legacy, hence was decided not worth it.

Does this make more sense? :)